Upgrade to Dokka 0.10.0

Dokka 0.10.0 was just [released](https://github.com/Kotlin/dokka/releases/tag/0.10.0), fixes a bunch of bugs and introduces multiplatform support.

This is currently blocked by https://github.com/vanniktech/gradle-maven-publish-plugin/issues/69 which tries to automatically apply the `dokka-android` plugin to `kotlin-android` modules.

Error message is:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':workflow-ui-android'.
> Failed to apply plugin [id 'org.jetbrains.dokka-android']
   > Cannot add task 'dokka' as a task with that name already exists.
```

--- a/kotlin/.buildscript/configure-dokka.gradle
+++ b/kotlin/.buildscript/configure-dokka.gradle
@@ -27,15 +27,17 @@ afterEvaluate { project ->
   project.tasks.dokka {
     outputDirectory = "$rootDir/../docs/kotlin/api"
     outputFormat = 'gfm'
-    reportUndocumented = false
-    skipDeprecated = true
-    jdkVersion = 8
-    packageOptions {
-      prefix = "com.squareup.workflow.internal"
-      suppress = true
-    }
-    if (project.file('Module.md').exists()) {
-      includes = ['Module.md']
+    configuration {
+      reportUndocumented = false
+      skipDeprecated = true
+      jdkVersion = 8
+      perPackageOption {
+        prefix = "com.squareup.workflow.internal"
+        suppress = true
+      }
+      if (project.file('Module.md').exists()) {
+        includes = ['Module.md']
+      }
     }
   }
 }

--- a/kotlin/build.gradle
+++ b/kotlin/build.gradle
@@ -84,8 +84,7 @@ buildscript {
               'mockito': "com.nhaarman:mockito-kotlin-kt1.1:1.6.0"
           ]
       ],
-      'dokkaAndroid': "org.jetbrains.dokka:dokka-android-gradle-plugin:0.9.18",
-      'dokka': "org.jetbrains.dokka:dokka-gradle-plugin:0.9.18",
+      'dokka': "org.jetbrains.dokka:dokka-gradle-plugin:0.10.0",
       'jmh': [
           'gradlePlugin': "me.champeau.gradle:jmh-gradle-plugin:0.5.0",
           'core': "org.openjdk.jmh:jmh-core:1.22",
@@ -93,7 +92,7 @@ buildscript {
       ],
       // Source of "API 'variant.getJavaCompile()' is obsolete" warning, fixed in upcoming 0.9.0.
       // See https://github.com/vanniktech/gradle-maven-publish-plugin/issues/67.
-      'mavenPublish': "com.vanniktech:gradle-maven-publish-plugin:0.8.0",
+      'mavenPublish': "com.vanniktech:gradle-maven-publish-plugin:0.9.0-SNAPSHOT",
       'ktlint': "org.jlleitschuh.gradle:ktlint-gradle:9.1.1",
       'lanterna': "com.googlecode.lanterna:lanterna:3.0.2",
       'detekt': "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.0.1",
@@ -127,7 +126,6 @@ buildscript {
     classpath deps.android_gradle_plugin
     classpath deps.detekt
     classpath deps.dokka
-    classpath deps.dokkaAndroid
     classpath deps.jmh.gradlePlugin
     classpath deps.kotlin.gradlePlugin
     classpath deps.ktlint
@@ -138,6 +136,10 @@ buildscript {
     mavenCentral()
     gradlePluginPortal()
     google()
+    // For the snapshot version of the maven publish plugin.
+    maven {
+      url { "https://oss.sonatype.org/content/repositories/snapshots" }
+    }
   }
 }
 

--- a/kotlin/workflow-ui-android/build.gradle
+++ b/kotlin/workflow-ui-android/build.gradle
@@ -15,7 +15,7 @@
  */
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
-apply plugin: 'org.jetbrains.dokka-android'
+apply plugin: 'org.jetbrains.dokka'
 
 sourceCompatibility = JavaVersion.VERSION_1_8
 targetCompatibility = JavaVersion.VERSION_1_8

